From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Black-Hole1 <bh@bugs.cc>
Date: Tue, 26 Sep 2023 10:00:44 +0800
Subject: refactor(packages): improve sshd config

allow root user login with ssh

diff --git a/package/openssh/openssh.mk b/package/openssh/openssh.mk
index e14d752290aab93cb055a68205ba66b1d8a98943..b10acc91454156d52ac86e391f3b211c18b8bbcf 100644
--- a/package/openssh/openssh.mk
+++ b/package/openssh/openssh.mk
@@ -113,6 +113,13 @@ endef
 OPENSSH_POST_INSTALL_TARGET_HOOKS += OPENSSH_INSTALL_SERVER_PROGRAMS
 
 define OPENSSH_INSTALL_INIT_SYSTEMD
+	$(SED) 's/\#PasswordAuthentication/PasswordAuthentication/' $(TARGET_DIR)/etc/ssh/sshd_config
+	$(SED) 's/\#PermitRootLogin.*/PermitRootLogin yes/' $(TARGET_DIR)/etc/ssh/sshd_config
+	$(SED) 's/\#MaxStartups.*/MaxStartups 500/' $(TARGET_DIR)/etc/ssh/sshd_config
+	$(SED) 's/\#MaxAuthTries.*/MaxAuthTries 200/' $(TARGET_DIR)/etc/ssh/sshd_config
+	$(SED) 's/\#MaxSessions.*/MaxSessions 100/' $(TARGET_DIR)/etc/ssh/sshd_config
+	$(SED) 's/\#ClientAliveCountMax.*/ClientAliveCountMax 30/' $(TARGET_DIR)/etc/ssh/sshd_config
+
 	$(INSTALL) -D -m 644 package/openssh/sshd.service \
 		$(TARGET_DIR)/usr/lib/systemd/system/sshd.service
 	$(OPENSSH_INSTALL_SYSTEMD_SYSUSERS)
