--- Make.Rules	2023-05-13 22:41:59.000000000 +0800
+++ ../Make.Rules	2023-09-15 14:23:46.600089894 +0800
@@ -63,14 +63,14 @@
 KERNEL_HEADERS := $(topdir)/libcap/include/uapi
 LIBCAP_INCLUDES = -I$(KERNEL_HEADERS) -I$(topdir)/libcap/include
 DEFINES := -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-SYSTEM_HEADERS = /usr/include
+SYSTEM_HEADERS ?= $(INCDIR)
 
 SUDO := sudo
-CC := $(CROSS_COMPILE)gcc
-LD := $(CC) -Wl,-x -shared
-AR := $(CROSS_COMPILE)ar
-RANLIB := $(CROSS_COMPILE)ranlib
-OBJCOPY := $(CROSS_COMPILE)objcopy
+CC ?= $(CROSS_COMPILE)gcc
+LD ?= $(CC) -Wl,-x -shared
+AR ?= $(CROSS_COMPILE)ar
+RANLIB ?= $(CROSS_COMPILE)ranlib
+OBJCOPY ?= $(CROSS_COMPILE)objcopy
 
 # Reference:
 #   CPPFLAGS used for building .o files from .c & .h files
@@ -121,7 +121,7 @@
 # Force enabled with "make DYNAMIC=yes ...".
 DYNAMIC := $(shell if [ ! -d "$(topdir)/.git" ]; then echo $(SHARED); else echo no ; fi)
 
-PAM_CAP ?= $(shell if [ -f /usr/include/security/pam_modules.h ]; then echo $(SHARED) ; else echo no ; fi)
+PAM_CAP = no
 
 # If your system does not support pthreads, override this as "no".
 #
