From 35228679f2a79004b3b9e46372d6c6690a1ecaa5 Mon Sep 17 00:00:00 2001
From: Graham Ollis <plicease@cpan.org>
Date: Sat, 27 Jun 2020 23:23:54 -0600
Subject: [PATCH 02/12] use $fh instead of $foo

Signed-off-by: Kevin Cui <bh@bugs.cc>
---
 Parser.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Parser.pm b/Parser.pm
index 20e6aab..2870b69 100644
--- a/Parser.pm
+++ b/Parser.pm
@@ -346,8 +346,8 @@ XML::Parser - A perl module for parsing XML documents
                    Default => \&other);
 
   open(my $fh, 'xmlgenerator |');
-  $p3->parse($foo, ProtocolEncoding => 'ISO-8859-1');
-  close($foo);
+  $p3->parse($fh, ProtocolEncoding => 'ISO-8859-1');
+  close($fh);
 
   $p3->parsefile('junk.xml', ErrorContext => 3);
 
-- 
2.43.0

