From b1d805cdd03a2f2d1e5ee17bf63249f502dc2c5c Mon Sep 17 00:00:00 2001
From: Samanta Navarro <ferivoz@riseup.net>
Date: Sat, 3 Oct 2020 11:50:36 +0000
Subject: [PATCH 03/12] Fix typo in documentation

Signed-off-by: Kevin Cui <bh@bugs.cc>
---
 Expat/Expat.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Expat/Expat.pm b/Expat/Expat.pm
index cb35445..bc6bbc8 100644
--- a/Expat/Expat.pm
+++ b/Expat/Expat.pm
@@ -1147,7 +1147,7 @@ The element declaration handlers are passed objects of this class as the
 content model of the element declaration. They also represent content
 particles, components of a content model.
 
-When referred to as a string, these objects are automagicly converted to a
+When referred to as a string, these objects are automagically converted to a
 string representation of the model (or content particle).
 
 =over 4
-- 
2.43.0

