From 41aef0551d213061fcebce84ac4668b0218a1774 Mon Sep 17 00:00:00 2001
From: Todd Rinaldo <toddr@cpan.org>
Date: Thu, 28 Dec 2023 19:55:00 +0000
Subject: [PATCH 08/12] Remove Expat/Makefile from sources and MANIFEST

Fixes #94

Signed-off-by: Kevin Cui <bh@bugs.cc>
---
 MANIFEST | 1 -
 1 file changed, 1 deletion(-)

diff --git a/MANIFEST b/MANIFEST
index 1e979b6..aade5db 100644
--- a/MANIFEST
+++ b/MANIFEST
@@ -2,7 +2,6 @@ Changes				Change log
 Expat/encoding.h		Header file; describes *.enc structure
 Expat/Expat.pm			XML::Parser::Expat module
 Expat/Expat.xs			Extension library
-Expat/Makefile
 Expat/Makefile.PL		perl MakeMaker script for XML::Parser::Expat
 Expat/typemap			XS typemap
 inc/Devel/CheckLib.pm
-- 
2.43.0

