From f2c53e6432a7b48eac85835f879912e6308f1417 Mon Sep 17 00:00:00 2001
From: Todd Rinaldo <toddr@cpan.org>
Date: Thu, 28 Dec 2023 20:13:54 +0000
Subject: [PATCH 09/12] Fix workflow

Signed-off-by: Kevin Cui <bh@bugs.cc>
---
 cpanfile | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 cpanfile

diff --git a/cpanfile b/cpanfile
new file mode 100644
index 0000000..8ecd1e2
--- /dev/null
+++ b/cpanfile
@@ -0,0 +1,4 @@
+use strict;
+use warnings;
+
+requires 'LWP::UserAgent' => 0;
-- 
2.43.0

