From 7096c4e45a62a23837382cf822d8c2dab780a3a1 Mon Sep 17 00:00:00 2001
From: Todd Rinaldo <toddr@cpan.org>
Date: Fri, 29 Dec 2023 00:28:08 +0000
Subject: [PATCH 12/12] Bump version to 2.47 for release to CPAN

Signed-off-by: Kevin Cui <bh@bugs.cc>
---
 Expat/Expat.pm | 2 +-
 Parser.pm      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Expat/Expat.pm b/Expat/Expat.pm
index bc6bbc8..cc1618f 100644
--- a/Expat/Expat.pm
+++ b/Expat/Expat.pm
@@ -7,7 +7,7 @@ use strict;
 use XSLoader;
 use Carp;
 
-our $VERSION = '2.46';
+our $VERSION = '2.47';
 
 our ( %Encoding_Table, @Encoding_Path, $have_File_Spec );
 
diff --git a/Parser.pm b/Parser.pm
index 2870b69..6af890e 100644
--- a/Parser.pm
+++ b/Parser.pm
@@ -16,7 +16,7 @@ use Carp;
 
 BEGIN {
     require XML::Parser::Expat;
-    $VERSION = '2.46';
+    $VERSION = '2.47';
     die "Parser.pm and Expat.pm versions don't match"
       unless $VERSION eq $XML::Parser::Expat::VERSION;
 }
-- 
2.43.0

